src/views/Team/Outlet.vue

```diff
@@ -329,30 +339,30 @@ export default defineComponent({
     segmentChanged(ev: CustomEvent) {
       this.state = ev.detail.value;
     },
-    fetchData() {
-      this.loading = true;
-      this.state = "about";
-      const slug: any = this.$route.params.teamSlug;
-      let promise;
-      if (this.store.getters.teamsBySlug[slug]) {
-        promise = Promise.resolve();
-      } else {
-        promise = this.store.dispatch("teams/fetch", slug);
-      }
+    async fetchData() {
+      try {
+        this.loading = true;
+        this.state = "about";
+        const slug: any = this.$route.params.teamSlug;
 
-      promise
-        .then(() => {
-          const teamPointer = this.store.getters.objectsMap[
-            this.store.getters.teamsBySlug[slug]
-          ].toPointer();
-          return Promise.all([
-            this.store.dispatch("teams/fetchNews", teamPointer),
-            this.store.dispatch("teams/fetchSubteams", teamPointer),
-          ]);
-        })
-        .then(() => {
-          this.loading = false;
-        });
+        if (!this.store.getters.teamsBySlug[slug])
+          await this.store.dispatch("teams/fetch", slug);
+
+        const teamPointer = this.store.getters.objectsMap[
+          this.store.getters.teamsBySlug[slug]
+        ].toPointer();
+
+        await Promise.all([
+          this.store.dispatch("teams/fetchNews", teamPointer),
+          this.store.dispatch("teams/fetchActivities", teamPointer),
+          this.store.dispatch("teams/fetchSubteams", teamPointer),
+        ]);
+
+        this.loading = false;
+      } catch (error) {
```

Comment:
what's this general `try-catch` about? unresolved/broken promises that aren't caught are shown in the log anyways, and so are any other execution problems...

Reply:
https://github.com/TeamFranka/affinity/pull/195